test/querystring.test.js

```diff
@@ -134,4 +134,42 @@ describe('JsonClient', function () {
             return done();
         });
     });
+
+
+    it('should prefer default query string in url over query options',
+    function (done) {
+        SERVER.get('/foo', function (req, res, next) {
+            assert.deepEqual(req.query, {
+                foo: 'baz'
+            });
+            res.send(200);
+            return next();
+        });
+
+        CLIENT = clients.createJsonClient({
```

Comment:
nit: there's another `CLIENT` variable that references another client in the enclosing scope, which seems like it could be a bit confusing.